xc/xenops_server_xen.ml

```diff
@@ -1514,7 +1514,8 @@ module VM = struct
     } in
     debug "static_max_mib=%Ld" static_max_mib;
     let pvinpvh_xen_cmdline =
-      let base =
+      let base = "pv-shim no-smep no-smap" in
```

Comment:
I assume these are hardcoded because you don't want these to be accidentally lost if someone overrides the cmdline in the platformdata. If we ever need to enable SMAP/SMEP can you set `smep=true smap=true` in the  platform data to create `no-smep no-smap smep=true smap=true` cmdline, i.e. would the last option win in Xen?

Reply:
This gets a bit confusing. The value `!pvinpvh_xen_cmdline` is set from the `xenopsd.conf` file, and has a default defined in `xenopsd.ml`. It can further be overridden on a per-VM basis by a platform key. We shouldn't hard-code some further keys right here.

I think that it is preferable if each of these just set the whole command line (apart from the shim-mem, which is computed below). Then you have straightforward and full control over what happens in both ways of overriding the defaults.

So I'd just put `let pvinpvh_xen_cmdline = ref "pv-shim no-smep no-smap console=xen"` in `xenopsd.ml`.